src/riak_ql_ddl_compiler.erl

```diff
@@ -109,24 +109,26 @@ compile({ok, ?DDL{} = DDL}) ->
     %% handle output directly from riak_ql_parser
     compile(DDL);
 compile(?DDL{ table = Table, fields = Fields } = DDL) ->
-    {ModName, Attrs, LineNo} = make_attrs(Table, ?LINENOSTART),
-    {VFns,         LineNo2}  = build_validn_fns(Fields,    LineNo),
-    {ACFns,        LineNo3}  = build_add_cols_fns(Fields,  LineNo2),
-    {ExtractFn,    LineNo4}  = build_extract_fn(DDL,  LineNo3),
-    {GetTypeFn,    LineNo5}  = build_get_type_fn([Fields], LineNo4, []),    
-    {GetPosnFn,    LineNo6}  = build_get_posn_fn(Fields,   LineNo5, []),
-    {GetPosnsFn,   LineNo7}  = build_get_posns_fn(Fields,  LineNo6, []),
-    {IsValidFn,    LineNo8}  = build_is_valid_fn(Fields,   LineNo7),
-    {DDLVersionFn, LineNo9}  = build_get_ddl_compiler_version_fn(LineNo8),
-    {GetDDLFn,     LineNo10} = build_get_ddl_fn(DDL, LineNo9, []),
-    {HashFns,      LineNo11} = build_identity_hash_fns(DDL, LineNo10),
-    {FieldOrdersFn, LineNo10} = build_field_orders_fn(DDL, LineNo9),
+    {ModName, Attrs,   LineNo} = make_attrs(Table, ?LINENOSTART),
+    {VFns,             LineNo2}  = build_validn_fns(Fields,    LineNo),
+    {ACFns,            LineNo3}  = build_add_cols_fns(Fields,  LineNo2),
+    {ExtractFn,        LineNo4}  = build_extract_fn(DDL,  LineNo3),
+    {GetTypeFn,        LineNo5}  = build_get_type_fn([Fields], LineNo4, []),    
+    {GetPosnFn,        LineNo6}  = build_get_posn_fn(Fields,   LineNo5, []),
+    {GetPosnsFn,       LineNo7}  = build_get_posns_fn(Fields,  LineNo6, []),
+    {IsValidFn,        LineNo8}  = build_is_valid_fn(Fields,   LineNo7),
+    {DDLVersionFn,     LineNo9}  = build_get_ddl_compiler_version_fn(LineNo8),
+    {GetDDLFn,         LineNo10} = build_get_ddl_fn(DDL, LineNo9, []),
+    {HashFns,          LineNo11} = build_identity_hash_fns(DDL, LineNo10),
+    {FieldOrdersFn,    LineNo10} = build_field_orders_fn(DDL, LineNo9),
     {RevertOrderingFn, LineNo11} = build_revert_ordering_on_local_key_fn(DDL, LineNo10),
-    {MinDDLCapFn, LineNo12} = build_min_ddl_version_fn(DDL, LineNo11),
+    {MinDDLCapFn,      LineNo12} = build_min_ddl_version_fn(DDL, LineNo11),
+    {DeleteKeyFn,      LineNo11} = build_delete_key_fn(DDL, LineNo10, []),
```

Comment:
You have caught the dirty secret the line numbers are only used in decompilation - but have fixed anyway ;-)